app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsViewModel.kt

```diff
@@ -16,19 +17,25 @@ class SimilarEventsViewModel(private val eventService: EventService) : ViewModel
     val progress = MutableLiveData<Boolean>()
     val similarEvents = MutableLiveData<List<Event>>()
     val error = MutableLiveData<String>()
+    var eventId: Long = -1
 
     fun loadSimilarEvents(id: Long) {
         if(id == -1L){
             return
         }
         compositeDisposable.add(eventService.getSimilarEvents(id)
                 .subscribeOn(Schedulers.io())
+                .switchMap {
+                    Observable.fromIterable(it)
+                        .filter{ it.id != eventId }
+                        .toList()
+                        .toFlowable()
+                }
```

Comment:
Not right

Reply:
toFlowable shoudn't be used ?

Reply:
The logic does satisfy here right as I am using a switch map to filter the event one by one and then converting it to a list and the list is then converted to a flowable. Is the logic still wrong ?

Reply:
I researched a bit an found out that the Flowable which is locked to the information returning from the Dao is not suitable for toList. toList requires a finite source. Converting Flowable to Single will be fruitful in this case.

2nd answer states this
https://stackoverflow.com/questions/50318141/rxjava-having-trouble-converting-one-list-to-another-list

Reply:
OK, then it would be more efficient to remove the item in subscribe method than create another stream

Reply:
Yup :+1: Let me try that

Reply:
Updated PR